DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs

```diff
@@ -149,68 +151,131 @@ private async Task FetchConfigAsyncWithTask()
+                            }
+                            else if (sseManager != null && localOptions.EnableBetaRealtimeUpdates)
+                            {
+                                sseManager.RestartSSE(sseUri);
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            logger.LogWarning(e, "Failed to parse SSE config. Skipping SSE Initialization");
+                        }
+
+                        localBucketing.StoreConfig(sdkKey, res.Content);
+                        configEtag = etag;
+                        configLastModified = lastModified;
+                        logger.LogDebug("Config successfully initialized with etag: {ConfigEtag}, {lastmodified}",
+                            configEtag, configLastModified);
+                        //eventQueue?.QueueSDKConfigEvent(request, res);
                     }
-                    
-                    localBucketing.StoreConfig(sdkKey, res.Content);
-                    configEtag = etag;
-                    configLastModified = lastModified;
-                    logger.LogDebug("Config successfully initialized with etag: {ConfigEtag}, {lastmodified}", configEtag, configLastModified);
-                    //eventQueue?.QueueSDKConfigEvent(request, res);
-                }
-                catch (Exception e)
-                {
-                    // This is to catch any exception that is thrown by the SetConfig method if the config is not valid
-                    logger.LogError($"Failed to set config: {e.Message} {e.InnerException.Message}");
-                }
+                    catch (Exception e)
+                    {
+                        // This is to catch any exception that is thrown by the SetConfig method if the config is not valid
+                        logger.LogError($"Failed to set config: {e.Message} {e.InnerException.Message}");
+                    }
+
+                    break;
+            }
+        }
+
+        private async void SSEMessageHandler(object sender, MessageReceivedEventArgs args)
+        {
+            var message = JsonSerializer.Deserialize<SSEMessage>(args.Message.Data);
+            if (message.Type is "refetchConfig" or "")
+            {
+                await FetchConfigAsyncWithTask(message.LastModified);
+            }
+        }
+
+        private void SSEErrorHandler(object sender, ExceptionEventArgs args)
+        {
+            logger.LogWarning(args.Exception, "SSE Connection Returned an error");
+        }
+
+        private void SSEStateHandler(object sender, StateChangedEventArgs args)
+        {
+            switch (args.ReadyState)
+            {
+                case ReadyState.Raw:
+                    break;
+                case ReadyState.Connecting:
+                    break;
+                case ReadyState.Open:
+                    var ssepollingIntervalMs = 60 * 60 * 15 * 1000;
```

Comment:
this should probably be a static const at the top?

Reply:
yeah - just currently messing around with some timings so i had it there - will move